views/src/Pages/ForgotPassword.jsx

```diff
@@ -0,0 +1,208 @@
+  const [emailSubmitted, setEmailSubmitted] = useState(false);
+  const [otpValidated, setOtpValidated] = useState(false);
+  const errorMsg = useRef(null);
+  const [otp, setOtp] = useState(0);
+  const [newPwd, setNewPwd] = useState(null);
+  const [confirmPwd, setConfirmPwd] = useState(null);
+  const updatepasswordValidator = (input, confirmation) => {
+    errorMsg.current.textContent = ``;
+    let newPwd = input;
+    let confirmPwd = confirmation;
+    if (!newPwd || !confirmPwd) {
+      errorMsg.current.textContent =
+        "Please enter and confirm your new password";
+    }
+    if (typeof newPwd !== "string") {
+      errorMsg.current.textContent = "New password must be a string";
+    }
+    if (newPwd !== confirmPwd) {
+      errorMsg.current.textContent = "Passwords do not match";
+    }
+    if (newPwd.length < 8) {
+      errorMsg.current.textContent =
+        "Password must be at least 8 characters long";
+    }
+
+    if (newPwd.length > 30) {
+      errorMsg.current.textContent =
+        "Password cannot be more than 30 characters long";
+    }
+  };
+  const nextStep = async () => {
+    try {
+      errorMsg.current.textContent = ``;
+      let response;
+      if (active === 0) {
+        const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+        // Make the API call to sendToken endpoint
+        if (emailRegex.test(email)) {
+          errorMsg.current.textContent = ``;
+          response = await axios.post(`${baseUrl}/api/password/forgotPassword`, { username: email });
+          if (response.status === 200) {
+            setEmailSubmitted(true);
+          } else if (response.status === 404) {
+            errorMsg.current.textContent = `Email not registered!`;
+          }
+        } else {
+          errorMsg.current.textContent = `Error! Enter valid email format.`;
+        }
+      } else if (active === 1) {
+        // Make the API call to check Token endpoint
+        console.log(otp, email);
+        response = await axios.post(`${baseUrl}/api/password/checkToken`, {
+          username: email,
+          resetToken: otp,
+        });
+        if (response.status === 200) {
+          setOtpValidated(true);
+        } else {
+          errorMsg.current.textContent = `Invalid OTP`;
+        }
+      }
+
+      // Check if response status is 200
+      if (response && response.status === 200) {
+        setActive((current) => (current < 3 ? current + 1 : current));
+      } else {
+        console.error("Error:", response.data.msg);
+      }
+    } catch (error) {
+      console.error("Error:", error);
+      if (error == "AxiosError: Request failed with status code 404") {
+        errorMsg.current.textContent = `Error: Email not registered!`;
+      }
+    }
+  };
+
+  const prevStep = () => {
+    errorMsg.current.textContent = ``;
+    setActive((current) => (current > 0 ? current - 1 : current));
+  };
+  const finishUp = async () => {
+    console.log(newPwd, confirmPwd);
```

Comment:
We don't want to have this console log here. Remove this

Reply:
Removed